### PR TITLE
Create Deploy Script For ForwardingStealthKeyResolver & Deploy To Rinkeby

### PR DESCRIPTION
**Pull request by apbendi**

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -1,4 +1,51 @@
...
+require("@nomiclabs/hardhat-etherscan");
+
+const chainIds = {
+  mainnet: 1,
+  rinkeby: 4,
+};
+
+// Ensure that we have all the environment variables we need.
+let mnemonic = '';
+if (!process.env.MNEMONIC) {
```
Comment (mds1) [inline hardhat.config.js:15]: Suggest adding a `.env.example` file to pull from

File: package.json [hunk 2]
```diff
@@ -24,10 +26,8 @@
   },
   "scripts": {
     "test": "hardhat test",
-    "watch": "npm-watch",
-    "lint": "solium --dir ./contracts",
-    "prepublishOnly": "truffle compile && truffle networks --clean",
-    "pub": "npm run prepublishOnly && npm publish"
+    "deploy": "hardhat compile && hardhat run scripts/deploy.js",
+    "lint": "solium --dir ./contracts"
```
Comment (mds1) [inline package.json:30]: Linting gives 14 errors and 51 warnings. For example it says `Syntax error: unexpected token a`, but everything compiles and tests run, so maybe solium is using the wrong solc version? Not sure if this is worth spending time on now

File: scripts/deploy.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+const hre = require("hardhat");
+const { exit } = require("process");
+const { ethers } = hre;
+
+const network = process.env.HARDHAT_NETWORK;
+
+// Initialize object that will hold all deploy info. We'll continually update this and save it to
+// a file using the save() method below
+const parameters = {
+  admin: null,
```
Comment (mds1) [inline scripts/deploy.js:11]: There's no admin right? I think this + `adminWallet` are just artifacts of our other deploy scripts. Just want to confirm

File: scripts/deploy.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// Setup for saving off deploy info to JSON files
+const now = new Date().toISOString();
+const folderName = "./deploy-history";
+const fileName = `${folderName}/${network}-${now}.json`;
+const latestFileName = `${folderName}/${network}-latest.json`;
+fs.mkdir(folderName, (err) => {
+  if (err && err.code !== "EEXIST") throw err;
+});
+
+// method to save the deploy info to 2 JSON files
```
Comment (mds1) [inline scripts/deploy.js:25]: This comment is wrong, only saves to 1 file

**Review by mds1 — CHANGES_REQUESTED** [review]
Looks great! Tests pass. Just a few small comments

File: scripts/deploy.js (hunk 3 above)
Comment (apbendi) [reply scripts/deploy.js:11]: Correct, no admin. That's just the deployer address being saved, though you're right, not great naming in this case.

File: package.json (hunk 2 above)
Comment (apbendi) [reply package.json:30]: Hmmm yeah it's barfing on Solidity stuff, i.e. the abstract keyword, so it must be linting for the wrong version. Will make an issue to look into it.

**Pull request merged**